lib/database/user.php

```diff
@@ -1374,7 +1374,7 @@ function getUserListByPerms($sortBy, $offset, $count, &$dataOut, $requestedBy, &
     $permsFilter = null;
 
     settype($perms, 'integer');
-    if ($perms >= Permissions::Spam && $perms <= Permissions::Unregistered || $perms == Permissions::SuperUser) {
+    if ($perms >= Permissions::Spam && $perms <= Permissions::Unregistered || $perms == Permissions::JrDeveloper) {
```

Comment:
This logic needs some thought. Currently, searching on `Registered` also returns all `Developer`s and `Admin`s. Because of this clause, it won't return `JrDeveloper`s. However, changing it would return `Developers`s and `Admin`s when searching for `JrDeveloper`s.

I think a better solution would be to have a specific clause for `Registered`, and make everything else only match the unique values.

Reply:
Additionally, if I'm reading things correctly, the only case that will ever get to line 1382 is `Root`.